debian/ubuntu-advantage-tools.postinst

```diff
@@ -95,45 +98,97 @@ if status:
 
 unconfigure_esm() {
     if ! check_service_is_enabled esm-infra; then
-        rm -f $APT_TRUSTED_KEY_DIR/ubuntu-esm*gpg  # Remove previous esm keys
-        rm -f $APT_TRUSTED_KEY_DIR/$ESM_INFRA_KEY_TRUSTY
-        rm -f $ESM_INFRA_APT_SOURCE_FILE_TRUSTY
-        rm -f $ESM_INFRA_OLD_APT_SOURCE_FILE_TRUSTY
-        rm -f $ESM_APT_PREF_FILE_TRUSTY $ESM_INFRA_OLD_APT_PREF_FILE_TRUSTY
-        rm -f $ESM_INFRA_APT_PREF_FILE_TRUSTY
+        rm -f "$APT_TRUSTED_KEY_DIR/ubuntu-esm*gpg"  # Remove previous esm keys
+        rm -f "$APT_TRUSTED_KEY_DIR/$ESM_INFRA_KEY_TRUSTY"
+        rm -f "$ESM_INFRA_APT_SOURCE_FILE"
+        rm -f "$ESM_INFRA_OLD_APT_SOURCE_FILE_TRUSTY"
+        rm -f "$ESM_APT_PREF_FILE_TRUSTY" "$ESM_INFRA_OLD_APT_PREF_FILE_TRUSTY"
+        rm -f "$ESM_INFRA_APT_PREF_FILE"
+    fi
+    if ! check_service_is_enabled esm-apps; then
+        rm -f "$APT_TRUSTED_KEY_DIR/$ESM_APPS_KEY"
+        rm -f "$ESM_APPS_APT_SOURCE_FILE"
+        rm -f "$ESM_APPS_APT_PREF_FILE"
     fi
 }
 
 
-configure_esm() {
-    rm -f $APT_TRUSTED_KEY_DIR/ubuntu-esm*gpg  # Remove previous esm keys
-    if [ ! -f "$APT_TRUSTED_KEY_DIR/$ESM_INFRA_KEY_TRUSTY" ]; then
-        cp $UA_KEYRING_DIR/$ESM_INFRA_KEY_TRUSTY $APT_TRUSTED_KEY_DIR
+# Add visibility to a disabled ESM APT source by installing a GPG key and
+# preferences file to Pin never so packages won't get installed by apt update.
+install_esm_apt_key_and_source() {
+    service=$1 release=$2
+    apt_suite="${release}-${service}";
+    case "${service}" in
+        apps)
+            apt_origin="UbuntuESMApps"
+            apt_pref_file=${ESM_APPS_APT_PREF_FILE};
+            apt_source_file=${ESM_APPS_APT_SOURCE_FILE};
+            apt_key=${ESM_APPS_KEY};
+            ;;
+        infra)
+            apt_origin="UbuntuESM"
+            apt_pref_file=${ESM_INFRA_APT_PREF_FILE};
+            apt_source_file=${ESM_INFRA_APT_SOURCE_FILE};
+            apt_key=${ESM_INFRA_KEY_TRUSTY};
```

Comment:
It seems odd to me that this key has "trusty" in the name but we are using it for all esm-infra, including xenial now.

Reply:
Yes totally agree. We had opted in this particular case to leave the key name unaltered in trusty release time-frame to avoid unnecessary risk (minimal) if the postinst renaming improperly. Historically APT repos originally were going to have a different gpg key to control the repo for each Ubuntu release. Subsequent iterations dropped that desire as unmaintainable.
We could rename here as well dropping the -trusty suffix, we would also have to make sure to handle upgrade paths for previous ua-tools installs to current and make sure we don't leave a stale gpg keyring artifact around across upgrade path.

Reply:
Ah that makes sense - thanks for the context. This is pretty minor, and I defer to you on whether its worth fitting in now or putting in the backlog for next release.

Reply:
Thanks @orndorffgrant I added tech-debt for https://github.com/canonical/ubuntu-advantage-client/issues/1539